File: apis/src/main/java/com/beat/apis/home/application/dto/HomeFindAllResponse.java
package com.beat.apis.home.application.dto;

import java.util.List;

public record HomeFindAllResponse(
	List<HomePromotionDetail> promotionList,
	List<HomePerformanceDetail> performanceList
) {
	public static HomeFindAllResponse of(List<HomePromotionDetail> promotionList,
		List<HomePerformanceDetail> performanceList) {
		return new HomeFindAllResponse(promotionList, performanceList);
	}
}

File: apis/src/main/kotlin/com/beat/apis/home/api/response/HomeFindAllResponse.kt
package com.beat.apis.home.api.response

import com.beat.apis.home.application.result.HomeFindAllResult

@ConsistentCopyVisibility
data class HomeFindAllResponse private constructor(
    val promotionList: List<HomePromotionDetail>,
    val performanceList: List<HomePerformanceDetail>,
) {
    companion object {
        fun from(result: HomeFindAllResult): HomeFindAllResponse = HomeFindAllResponse(
            promotionList = result.promotionList.map(HomePromotionDetail::from),
            performanceList = result.performanceList.map(HomePerformanceDetail::from),
        )
    }
}
